Update component drawings

weerox opened the pull request:
The Buffer, And, Or and Xor gates has been given proper toolbuttons,
the component drawing has been updated, the toolbutton image matches
the component drawing and the I/O ports are now drawn according to the
vector values of GetPorts().

weerox commented:
There is still some work left to do on the icons and drawings, the buffer should be a little smaller and the Not gate has to be updated (both the GetPorts() thing and the look of it to match the buffer), but maybe we could still merge this and I'll create some issues?

NogginBops commented on src/LogikUI/Simulation/Gates/AndGate.cs:
```diff
@@ -17,9 +17,9 @@ class AndGate : IComponent
-            ports[2] = new Vector2i(-3, 1);
+            ports[0] = new Vector2i(-1, 1);
+            ports[1] = new Vector2i(-1, -1);
+            ports[2] = new Vector2i(2, 0);
```
This is not correct. The output of the component should be at the mouse position. It's much harder to place components when the mouse is centered around the mouse instead of at the output port

NogginBops commented on src/LogikUI/Simulation/Gates/BufferGate.cs:
```diff
@@ -0,0 +1,60 @@
+        public void GetPorts(Span<Vector2i> ports)
+        {
+            ports[0] = new Vector2i(-1, 0);
+            ports[1] = new Vector2i(2, 0);
```
Once again the output should be at the origin.

NogginBops commented on src/LogikUI/Simulation/Gates/OrGate.cs:
```diff
@@ -15,51 +15,42 @@ class OrGate : IComponent
-            ports[2] = new Vector2i(-3, 1);
+            ports[0] = new Vector2i(-1, 1);
+            ports[1] = new Vector2i(-1, -1);
+            ports[2] = new Vector2i(2, 0);
```
Output at origin.

NogginBops commented on src/LogikUI/Simulation/Gates/XorGate.cs:
```diff
@@ -0,0 +1,62 @@
+        {
+            ports[0] = new Vector2i(-1, 1);
+            ports[1] = new Vector2i(-1, -1);
+            ports[2] = new Vector2i(2, 0);
```
Output at origin.

weerox commented:
Yep, I'll fix that, I did feel a little weird having the mouse float in the center of the component.

weerox commented:
I also now realize that is how it works in the schematics software I've been using, weird that I didn't catch that before.

weerox commented:
@NogginBops I'm fixing the mouse origin now and I noticed that orientating a component might be more intuitive without PR #33. When I changed that (which was when the mouse origin was at the center of the component) I thought that it'd be more intuitive to have the component "point" with the output pin in the direction I was pressing, i.e. left arrow should have the component pointing to the left, but now that I've changed the origin to be on the output pin, I'm tempted to orient the component as "on this side of the mouse pointer", i.e. left arrow should switch the component to the left side of the mouse pointer (pointing to the right). 

What do you think? 
What should happen when we add components that doesn't look like it is pointing in a certain direction (i.e. Flip-Flops)? (This is what makes me think that the "left arrow -> point right" might be the way to go)

NogginBops commented:
@weerox I'm used to the way logisim does it where the direction you pick is where the connection point at the mouse should be pointing to the direction you pressed.

I don't know if one way or the other is more intuitive, but if nothing else, people that have used logisim will be used to the way we have atm so that can be an argument for doing it that way (We could provide an option for it though).

Here is an illustration of how the direction thing works for more complex components (green ring is mouse pos with direction):
![image](https://user-images.githubusercontent.com/3778357/81177696-60871f80-8fa7-11ea-8f21-e32633bc6cba.png)

NogginBops commented on src/LogikUI/Program.cs:
```diff
@@ -29,23 +29,21 @@ class Program
+
             ComponentTool andTool = new ComponentTool(ComponentType.And, "And gate", editor, toolbar);
 
-            ComponentTool notTool = new ComponentTool(ComponentType.Not, "Not gate", editor, toolbar);
```
What happened to the not gate tool?

weerox commented:
In both those images you've pressed the right arrow?

weerox commented:
I removed the NOT gate and replaced it with the Buffer, didn't we talk about having the NOT gate as the not-version of the buffer (as with AND and NAND) when pressing `n`?

NogginBops commented:
Yeah we said that the should be the case but I still think that the not-gate should be in the toolbar because that is a lot more common to use than a buffer. So if anything we should switch the buffer in the toolbar to be a not gate.

weerox commented:
I have updated the component origins to be at the output port and created an issue for swapping the buffer for a NOT gate. (The reason why I'd like to postpone the NOT gate swap is that I'd like to create all of the NOT versions at the same time, which might take a while)

weerox commented:
Also, just to be completely sure, does the orientation in the `master` branch behave as you'd expect it to, or should they be flipped?

NogginBops commented:
The orientations in the master branch should be correct iirc. 
Will check this pr later today.

NogginBops commented:
Great pr! Merging.
We still want to make the buffer gate smaller imo but we can do that in another pr.

weerox commented:
Yeah, I mentioned a couple of other things above that also has to be changed, I'll create a couple of issues.
